/*
 * Copyright 2024 Sowers, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

option go_package = "bosca.io/api/protobuf/search";

import "google/api/annotations.proto";

package bosca.ai;

service AIService {
  rpc QueryStorage(QueryStorageRequest) returns (QueryResponse) {
    option (google.api.http) = {
      post: "/v1/ai/query"
      body: "*"
    };
  }

  rpc QueryPrompt(QueryPromptRequest) returns (QueryResponse) {
    option (google.api.http) = {
      post: "/v1/ai/prompt"
      body: "*"
    };
  }
}

message QueryStorageRequest {
  string storage_system = 1;
  string query = 2;
}

message QueryPromptRequest {
  string prompt_id = 1;
  string model_id = 2;
  map<string, string> arguments = 3;
}

message QueryResponse {
  string response = 1;
}